projects/packages/publicize/src/class-publicize.php

```diff
@@ -259,7 +259,7 @@ public function get_all_connections_for_user() {
 								array(
 									'service_name'   => $service_name,
 									'connection_id'  => $connection['connection_data']['id'],
-									'can_disconnect' => current_user_can( 'manage_options' ) || get_current_user_id() === $user_id,
+									'can_disconnect' => current_user_can( 'edit_others_posts' ) || get_current_user_id() === $user_id,
```

Comment:
This is the permission that [Calypso uses](https://github.com/Automattic/wp-calypso/blob/c1cb83c9aa061c8f7b197c97bbe19d1c2ad0c19c/client/state/sharing/publicize/selectors.js#L87).